packages/perf/src/main.ts

```diff
@@ -0,0 +1,132 @@
+    'run-server': {
+      type: 'boolean',
+      demandOption: true,
+      default: false,
+      description: 'Whether to run as a server'
+    },
+    'server-address': {
+      type: 'string',
+      demandOption: false,
+      description: 'Server IP address',
+      default: ''
+    },
+    transport: {
+      type: 'string',
+      demandOption: false,
+      description: 'Transport to use',
+      default: 'tcp'
+    },
+    'upload-bytes': {
+      type: 'number',
+      demandOption: false,
+      description: 'Number of bytes to upload',
+      default: 0
+    },
+    'download-bytes': {
+      type: 'number',
+      demandOption: false,
+      description: 'Number of bytes to download',
+      default: 0
+    }
+  })
+  .command('help', 'Print usage information', yargs.help)
+  .parseSync()
+
+export async function main (runServer: boolean, serverIpAddress: string, transport: string, uploadBytes: number, downloadBytes: number): Promise<void> {
+  const listenAddrs: string[] = []
+
+  const { host, port } = splitHostPort(serverIpAddress)
+  // #TODO: right now we only support tcp
+  const tcpMultiaddr = multiaddr(`/ip4/${host}/tcp/${port}`)
+
+  const config = {
+    transports: [tcp()],
+    streamMuxers: [yamux()],
+    connectionEncryption: [
+      plaintext()
+    ],
+    services: {
+      perf: perfService(defaultInit)
+    }
+  }
+
+  const testPrivKey = 'CAASpgkwggSiAgEAAoIBAQC2SKo/HMFZeBml1AF3XijzrxrfQXdJzjePBZAbdxqKR1Mc6juRHXij6HXYPjlAk01BhF1S3Ll4Lwi0cAHhggf457sMg55UWyeGKeUv0ucgvCpBwlR5cQ020i0MgzjPWOLWq1rtvSbNcAi2ZEVn6+Q2EcHo3wUvWRtLeKz+DZSZfw2PEDC+DGPJPl7f8g7zl56YymmmzH9liZLNrzg/qidokUv5u1pdGrcpLuPNeTODk0cqKB+OUbuKj9GShYECCEjaybJDl9276oalL9ghBtSeEv20kugatTvYy590wFlJkkvyl+nPxIH0EEYMKK9XRWlu9XYnoSfboiwcv8M3SlsjAgMBAAECggEAZtju/bcKvKFPz0mkHiaJcpycy9STKphorpCT83srBVQi59CdFU6Mj+aL/xt0kCPMVigJw8P3/YCEJ9J+rS8BsoWE+xWUEsJvtXoT7vzPHaAtM3ci1HZd302Mz1+GgS8Epdx+7F5p80XAFLDUnELzOzKftvWGZmWfSeDnslwVONkL/1VAzwKy7Ce6hk4SxRE7l2NE2OklSHOzCGU1f78ZzVYKSnS5Ag9YrGjOAmTOXDbKNKN/qIorAQ1bovzGoCwx3iGIatQKFOxyVCyO1PsJYT7JO+kZbhBWRRE+L7l+ppPER9bdLFxs1t5CrKc078h+wuUr05S1P1JjXk68pk3+kQKBgQDeK8AR11373Mzib6uzpjGzgNRMzdYNuExWjxyxAzz53NAR7zrPHvXvfIqjDScLJ4NcRO2TddhXAfZoOPVH5k4PJHKLBPKuXZpWlookCAyENY7+Pd55S8r+a+MusrMagYNljb5WbVTgN8cgdpim9lbbIFlpN6SZaVjLQL3J8TWH6wKBgQDSChzItkqWX11CNstJ9zJyUE20I7LrpyBJNgG1gtvz3ZMUQCn3PxxHtQzN9n1P0mSSYs+jBKPuoSyYLt1wwe10/lpgL4rkKWU3/m1Myt0tveJ9WcqHh6tzcAbb/fXpUFT/o4SWDimWkPkuCb+8j//2yiXk0a/T2f36zKMuZvujqQKBgC6B7BAQDG2H2B/ijofp12ejJU36nL98gAZyqOfpLJ+FeMz4TlBDQ+phIMhnHXA5UkdDapQ+zA3SrFk+6yGk9Vw4Hf46B+82SvOrSbmnMa+PYqKYIvUzR4gg34rL/7AhwnbEyD5hXq4dHwMNsIDq+l2elPjwm/U9V0gdAl2+r50HAoGALtsKqMvhv8HucAMBPrLikhXP/8um8mMKFMrzfqZ+otxfHzlhI0L08Bo3jQrb0Z7ByNY6M8epOmbCKADsbWcVre/AAY0ZkuSZK/CaOXNX/AhMKmKJh8qAOPRY02LIJRBCpfS4czEdnfUhYV/TYiFNnKRj57PPYZdTzUsxa/yVTmECgYBr7slQEjb5Onn5mZnGDh+72BxLNdgwBkhO0OCdpdISqk0F0Pxby22DFOKXZEpiyI9XYP1C8wPiJsShGm2yEwBPWXnrrZNWczaVuCbXHrZkWQogBDG3HGXNdU4MAWCyiYlyinIBpPpoAJZSzpGLmWbMWh28+RJS6AQX6KHrK1o2uw=='
```

Comment:
Is hard coding the key part of the protocol?

Reply:
Not explicitly, but I want to have a stable identity i.e. the`peerID`  and given that it's not passed in as an argument, I figured this was the best way to have the client connect to the server without having to do a prior exchange of the server's peerID.

Reply:
Unfortunately the libp2p perf runner does not have support for the server to send its peer ID to the client.

In go-libp2p we simply hardcode it:

https://github.com/libp2p/test-plans/blob/a674f734cc625177a91aea84f568a50cfd70ab7d/perf/impl/go-libp2p/v0.29/main.go#L37-L42

rust-libp2p allows connecting to an address without its peer ID and thus does not require hard-coding.

Adding support for the server to send its peer ID to the client is tracked in https://github.com/libp2p/test-plans/issues/63. That said, I think there are higher impact items to tackle beforehand.